pkg/kubelet/winstats/perfcounters.go

```diff
@@ -0,0 +1,120 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winstats
+
+import (
+	"errors"
+	"fmt"
+	"github.com/lxn/win"
+	"time"
+	"unsafe"
+)
+
+const (
+	CPUQuery                  = "\\Processor(_Total)\\% Processor Time"
+	MemoryPrivWorkingSetQuery = "\\Process(_Total)\\Working Set - Private"
+	MemoryCommittedBytesQuery = "\\Memory\\Committed Bytes"
+)
+
+// Metric collected from a counter
+type Metric struct {
+	Name  string
+	Value float64
+}
+
+// NewMetric construct a Metric struct
+func NewMetric(name string, value float64) Metric {
+	return Metric{
+		Name:  name,
+		Value: value,
+	}
+}
+
+func (metric Metric) String() string {
+	return fmt.Sprintf(
+		"Name: %s | Value: %s",
+		metric.Name,
+		metric.Value,
+	)
+}
+
+func readPerformanceCounter(counter string, sleepInterval int) (chan Metric, error) {
+
+	var queryHandle win.PDH_HQUERY
+	var counterHandle win.PDH_HCOUNTER
+
+	ret := win.PdhOpenQuery(0, 0, &queryHandle)
+	if ret != win.ERROR_SUCCESS {
+		return nil, errors.New("Unable to open query through DLL call")
+	}
+
+	// test path
+	ret = win.PdhValidatePath(counter)
+	if ret == win.PDH_CSTATUS_BAD_COUNTERNAME {
```

Comment:
You also need to handle other errors too (https://msdn.microsoft.com/en-us/library/windows/desktop/aa372986(v=vs.85).aspx).